_layouts/home.html

```diff
@@ -264,6 +264,18 @@ <h3>Legend</h3>
      * @returns {string}
      */
     function popupTemplate(shelter) {
+
+        // Takes care of null here
+        let shelterPets = shelter.pets_notes;
+        if (shelterPets === 'null' || shelterPets === null) {
+            shelterPets = 'Unknown';
```

Comment:
There's also a boolean (I think, or three-state?) value `shelters.pets` which says whether pets are allowed at all. For the `shelter.pets_notes` field, I think the intent was for this to provide additional detail if we have it.

@omnilord can you confirm the `shelters.pets` value? Is it two-state or three-state?

Assuming that's correct, then because of that, `'Unknown'` isn't really appropriate -- I think (as with `shelter.notes` below) an empty string `''` default would be better.

Reply:
To be a bit more clear here, I think `shelter.pets` should be rendered as `true => 'Yes'`, `false => 'No'`, `null => 'Unknown'`. Or we should just make that a three-state value accordingly in the API.

The `shelter.pets_notes` field should be empty or have info as needed.

Reply:
I went off of the previous code. In the previous code, `shelter.pets_note` is what was used to put in any information about pets so that's what I used. I don't know if it was the actual status of whether pets are allowed or if it was meant as an extra note, so I can't help there.

I think it might make more sense to keep this PR focused on the issue as it was written. If we want to mess around with what properties are sent from further upstream or what the names of properties are there, that should be turned into its own issue.

As far as empty fields go, I've found they generally cause confusion. There should be some indication of status for every field.

Just fyi, the values come in as strings. `null` comes in as `'null'`. I was prematurely future-proofing when I included `=== null`.

Reply:
Are you suggesting we close the issue then and have it rewritten? The goal here is to concisely display accurate data, not semantics of issues and PRs. We are live and advertising the map. Notes are intended to be that, extra info.  The NSS doesn't have a field for pets and we are extending it. Every county is different so it's difficult to have blanket info and we have tried to adapt to a variety of scenarios.

Reply:
OK, I'd like to wait and have @omnilord weigh in here on the `shelter.pets` field. It may be a better field to use for this.

In my opinion, any text field value that's a "notes" field intended to provide additional detail should just not be there at all (i.e., no header label and no value) in the front-end if it's empty/null, rather than having a placeholder value like `'None'` or `'Unknown'`.

Reply:
I definitely agree we should *not* be displaying "null" to end users, so I think we're in agreement there.

Reply:
Definitely in agreement about 'null'.

My suggestion here is that we do small iterations. Our teams have found that to be a very useful pattern. We would finish with this issue first. We would make a new issue to deal with the deeper questions here. That would also have the advantage of getting this conversation into an issue. I've found issue conversations easier to look back on than conversations in PRs.

UX-wise, I've found people don't trust information as much when it's not presented consistently. I believe from what I've seen, that having a field of 'Unknown' is more reassuring than having a missing field.

If we want someone to weigh in on that, I'd suggest someone with user research/UX design experience.

Reply:
OK, per conversation in Slack with @omnilord --

> Omnilord 12:00 PM
There is a field `allow_pets` that is a boolean, and `pets` is a text field.
ryan.b.harvey (he/him):code_for_nola: 12:01 PM
OK, is that separate from `pets_notes`?
Or are things renamed in the JSON export somehow?
Omnilord 12:01 PM
There is no `pets_notes` column in the database.  I’m looking up the controller now
```
json.pets shelter.allow_pets ? 'Yes' : 'No'
json.pets_notes shelter.pets
```
ryan.b.harvey (he/him):code_for_nola: 12:02 PM
Yep, so renamed

So, based on that, I think we should display the `shelter.pets` field, and if there is data in `shelter.pets_notes`, that should be displayed as an addition to the value in `shelter.pets`.

Reply:
I agree with small iterations. I'm just concerned that this is the wrong fix for this problem, which is why I'm asking these questions and making these suggestions.

We already have code to exclude those fields entirely if there's no data in them. I think that's failing due to the `null` value coming in as a string. Perhaps a better fix is to take a look at [these conditionals](https://github.com/hurricane-response/response-theme/blob/71f9449916897c5073872dd52d8ade84b2843976/_layouts/home.html#L333-L334) and make sure they're doing what's intended?

Reply:
The front end is getting `.pets` and `.pets_notes` on `shelter`

Reply:
I suppose I'm coming from a place where the issue being in this repo (instead of in the API) was always meant as a temporary fix.

Also, fields sometimes not existing sounds a bit concerning.

Reply:
I also respect small iterations. And when not in an active response, that should be the default. But we are live and are looking for a compromise that solves the immediate problem in the quickest, best way now.